ansible/module_utils/port_utils.py

```diff
@@ -221,10 +221,10 @@ def get_port_alias_to_name_map(hwsku, asic_name=None):
 
 
 def get_port_indices_for_asic(asic_id, port_name_list_sorted):
-    front_end_port_name_list = [p for p in port_name_list_sorted if 'BP' not in p]
-    back_end_port_name_list = [p for p in port_name_list_sorted if 'BP' in p]
+    front_end_port_name_list = [p for p in port_name_list_sorted if 'BP' not in p and 'IB' not in p and 'Rec' not in p]
```

Comment:
Hi @sanmalho-git,
As I understand in [ get_port_alias_to_name_map](https://github.com/Azure/sonic-mgmt/blob/3cc50dcc7d25ebc4ee33baafaf3f009bc32aa175/ansible/module_utils/port_utils.py#L15) if we dont support using port_config.ini, the  port_alias_to_name_map is generated for only the frontend ports. how are the inb/rec ports are added to this map?

Reply:
the multi_asic.get_port_table(namespace='asic0') when executed on our linecard returns both the inband and recycle ports.
```
>>> 
admin@ixre-egl-board7:~$ python
Python 3.9.2 (default, Feb 28 2021, 17:03:44) 
[GCC 10.2.1 20210110] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from swsscommon import swsscommon
>>> from sonic_py_common import multi_asic
>>> swsscommon.SonicDBConfig.load_sonic_global_db_config()
>>> print(multi_asic.get_port_table(namespace='asic0').keys())
dict_keys(['Ethernet-IB0', 'Ethernet-Rec0', 'Ethernet0', 'Ethernet1', 'Ethernet10', 'Ethernet11', 'Ethernet12', 'Ethernet13', 'Ethernet14', 'Ethernet15', 'Ethernet16', 'Ethernet17', 'Ethernet2', 'Ethernet3', 'Ethernet4', 'Ethernet5', 'Ethernet6', 'Ethernet7', 'Ethernet8', 'Ethernet9'])

```

I am not sure, but this could be because the role defined for the inband and recylce ports is 'Inb' and 'Rec' - and not 'Int' in the port_config.ini. This change was requested by Judy.  Our port_config.ini file for asic0:

```
admin@ixre-egl-board7:~$ cat /usr/share/sonic/device/x86_64-nokia_ixr7250e_36x400g-r0/Nokia-IXR7250E-36x400G/0/port_config.ini 
# name         lanes                               alias            index  asic_port_name role  speed    coreid  coreportid  numvoq
Ethernet0      72,73,74,75,76,77,78,79             Ethernet0        1      Eth0-ASIC0     Ext   400000   1       1           8
Ethernet1      80,81,82,83,84,85,86,87             Ethernet1        2      Eth1-ASIC0     Ext   400000   1       2           8
Ethernet2      88,89,90,91,92,93,94,95             Ethernet2        3      Eth2-ASIC0     Ext   400000   1       3           8
Ethernet3      96,97,98,99,100,101,102,103         Ethernet3        4      Eth3-ASIC0     Ext   400000   1       4           8
Ethernet4      104,105,106,107,108,109,110,111     Ethernet4        5      Eth4-ASIC0     Ext   400000   1       5           8
Ethernet5      112,113,114,115,116,117,118,119     Ethernet5        6      Eth5-ASIC0     Ext   400000   1       6           8
Ethernet6      120,121,122,123,124,125,126,127     Ethernet6        7      Eth6-ASIC0     Ext   400000   1       7           8
Ethernet7      128,129,130,131,132,133,134,135     Ethernet7        8      Eth7-ASIC0     Ext   400000   1       8           8
Ethernet8      136,137,138,139,140,141,142,143     Ethernet8        9      Eth8-ASIC0     Ext   400000   1       9           8
Ethernet9      64,65,66,67,68,69,70,71             Ethernet9        10     Eth9-ASIC0     Ext   400000   0       10          8
Ethernet10     56,57,58,59,60,61,62,63             Ethernet10       11     Eth10-ASIC0    Ext   400000   0       11          8
Ethernet11     48,49,50,51,52,53,54,55             Ethernet11       12     Eth11-ASIC0    Ext   400000   0       12          8
Ethernet12     40,41,42,43,44,45,46,47             Ethernet12       13     Eth12-ASIC0    Ext   400000   0       13          8
Ethernet13     32,33,34,35,36,37,38,39             Ethernet13       14     Eth13-ASIC0    Ext   400000   0       14          8
Ethernet14     24,25,26,27,28,29,30,31             Ethernet14       15     Eth14-ASIC0    Ext   400000   0       15          8
Ethernet15     16,17,18,19,20,21,22,23             Ethernet15       16     Eth15-ASIC0    Ext   400000   0       16          8
Ethernet16     8,9,10,11,12,13,14,15               Ethernet16       17     Eth16-ASIC0    Ext   400000   0       17          8
Ethernet17     0,1,2,3,4,5,6,7                     Ethernet17       18     Eth17-ASIC0    Ext   400000   0       18          8
Ethernet-IB0   115                                 Ethernet-IB0     37     Rcy1-ASIC0     Inb   10000    0       19          8
Ethernet-Rec0  116                                 Ethernet-Rec0    39     Rcy2-ASIC0     Rec   10000    1       20          8

```